### PR TITLE
[ng] router chapter

### PR DESCRIPTION
**Pull request by chalin**
- Ignore the `angular/guide/router.html` as that is obsolete (it will be removed shortly, after I've finished extracting what is still useful from the file).

Contributes to #198

### CODE REVIEW

File: src/_data/side-nav.yml [hunk 1]
```diff
@@ -51,6 +51,19 @@
         - title: 6. HTTP
           permalink: /angular/tutorial/toh-pt6
 
+    - title: Router
```
Comment (kwalrath) [inline src/_data/side-nav.yml]: It seems weird to have Router at the same level as Guide, Tutorial, and Advanced. Although... it might be defensible if we put Angular-related packages (such as angular2_components) in the same place, since the router will also be separate (right?). But maybe even then, they should go under WEB LIBRARIES?

For now, I'm inclined to have the first page (Routing & Navigation) either in the guide, after Template Syntax, or in Advanced. The remaining pages would be in Advanced, grouped together somehow so that you don't see all of them unless you're interested.

File: src/angular/router/index.jade [hunk 2]
```diff
@@ -0,0 +1,312 @@
+---
+layout: angular
+title: Angular Router
```
Comment (kwalrath) [inline src/angular/router/index.jade]: Why call this "Angular Router" instead of "Routing & Navigation" or just "Routing"?

File: src/angular/router/index.jade [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+description: Discover the basics of screen navigation with the Angular Router
```
Comment (kwalrath) [inline src/angular/router/index.jade]: Why "Angular Router" instead of "Angular router"?

File: src/angular/router/index.jade [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+nextpage:
+  title: 1. Crisis Center App Starter
```
Comment (kwalrath) [inline src/angular/router/index.jade]: I know there's precedent, but I dislike having numbers in titles unless the number really matters and is labeled (e.g. "Step 1").

File: src/_data/side-nav.yml [hunk 5]
```diff
@@ -51,6 +51,19 @@
...
+      permalink: /angular/router
+      children:
+        - title: Introduction
+          permalink: /angular/router
+          urlExactMatch: true
+        - title: 1. App Starter
```
Comment (kwalrath) [inline src/_data/side-nav.yml]: We should name pages after user tasks, rather than the examples they feature.

File: src/angular/router/crisis-center-1.jade [hunk 6]
```diff
@@ -0,0 +1,294 @@
+---
+layout: angular
+title: 1. Crisis Center App Starter
+description: Getting started with the router
+prevpage:
+  title: Angular Router
+  url: /angular/router/index
```
Comment (kwalrath) [inline src/angular/router/crisis-center-1.jade]: Should `index` really be part of the URL?

File: src/angular/router/index.jade [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+  url: /angular/router/crisis-center-1
+---
+//- FilePath: src/angular/router/index.jade
+include ../_util-fns
+- var _example = 'router';
+
+//- TODO: sweep this chapter and replace "[an] array" by "[a] list".
+
+:marked
+  The Angular **Router** enables navigation from one [view](../glossary.html#view) to the next
```
Comment (kwalrath) [inline src/angular/router/index.jade]: Why the "Angular..." prefix, and why capital R Router?

Comment (kwalrath) [inline src/angular/router/index.jade]: (This is a global comment.)

File: src/_data/side-nav.yml (hunk 1 above)
Comment (chalin) [inline src/_data/side-nav.yml]: The side nav depth is limited to 2 levels (+ a possible section heading) for good reason, but it does force us to give some extra though to the docs organization.

Yes, we will inevitably want a way to be able to document angular relevant libraries and packages separately, if only because we cannot fit pack it all into two levels of the side nav.

I think that a Router/Routing section fits well where it is place, though maybe I can move it to after "Advanced".

File: src/angular/router/index.jade (hunk 2 above)
Comment (chalin) [inline src/angular/router/index.jade]: I was looking for something short. "Routing" is a good candidate, though we cover more than routing, this new guide describes the router features.

File: src/angular/router/index.jade (hunk 3 above)
Comment (chalin) [inline src/angular/router/index.jade]: "Router" is taken as a proper name (as in "the Router library"). This is who it was written in the TS docs and, as I mentioned elsewhere, I chose to focus on getting core content done, thinking that we can come back to tweak capitalization as needed.

File: src/angular/router/index.jade (hunk 4 above)
Comment (chalin) [inline src/angular/router/index.jade]: I agree. The titles used to be "Milestone 1. ..." but at the same time we want to keep titles short. I.e., there is a balance to be had; cf. #175.

File: src/angular/router/index.jade [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+  The Angular Router is not part of the Angular core.
+  It is an optional service packaged within its own library, `angular2.router`.
+  Import it like any other package.
+
++makeExcerpt('lib/app_component_1.dart', 'import', '')
+
+:marked
+  ### Configuration
+
+  When the browser's URL changes, the router looks for a corresponding **`RouteDefinition`**
```
Comment (chalin) [inline src/angular/router/index.jade]: I believe that most of the time bold is used to mark the point of first use of the named class. Do you still want me to remove the bold (this convention is also used in other chapters).

File: src/angular/router/index.jade [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
++makeExcerpt('app/app.component.1.ts (routes)', 'app-component-routes')
+
+:marked
+  There are several flavors of `RouteDefinition`.
+  The most common, illustrated above, is the named **`Route`** which maps a URL path to a component.
+
+:marked
+  ### Router outlet
+
+  Given this configuration, when the browser URL for this application becomes `/heroes`,
```
Comment (chalin) [inline src/angular/router/index.jade]: Fixed

File: src/angular/router/index.jade [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+      a route. Clicking an element with a <code>routerLink</code> directive
+      that is bound to a <i>link parameters list</i> triggers a navigation.
+  tr
+    td <b><i>Link Parameters List</i></b>
+    td.
+      A list that the router interprets as a routing instruction.
+      You can bind that list to a <code>RouterLink</code> or pass the list as an argument to
+      the <code>Router.navigate</code> method.
+  tr
+    td <b><i>Routing Component</i></b>
```
Comment (chalin) [inline src/angular/router/index.jade]: Agreed

File: src/angular/router/index.jade [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+    td.
+      An Angular component with a <code>RouterOutlet</code> that displays views based on router navigations.
+
+.l-main-section
+:marked
+  ## The Sample Application
```
Comment (chalin) [inline src/angular/router/index.jade]: fixed

File: src/angular/router/index.jade [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+  Imagine an application that helps the _Hero Employment Agency_ run its business.
+  Heroes need work and the agency finds crises for them to solve.
+
+  The application has three main features:
+
+  1. A *Crisis Center* for maintaining the list of crises for assignment to heroes.
+  1. A *Heroes* area for maintaining the list of heroes employed by the agency.
+  1. An *Admin* area to manage the list of crises and heroes.
+
+  Try it by clicking on this <live-example title="Hero Employment Agency Live Example">live example link</live-example>.
```
Comment (chalin) [inline src/angular/router/index.jade]: Ok

File: src/angular/router/index.jade [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+    td.
+      The directive (<code>&lt;router-outlet></code>) that marks where the router should display a view.
+  tr
+    td <code>RouterLink</code>
+    td.
+      The directive for binding a clickable HTML element to
+      a route. Clicking an element with a <code>routerLink</code> directive
+      that is bound to a <i>link parameters list</i> triggers a navigation.
+  tr
+    td <b><i>Link Parameters List</i></b>
```
Comment (chalin) [inline src/angular/router/index.jade]: I've switched to sentence case.

File: src/angular/router/index.jade (hunk 10 above)
Comment (chalin) [inline src/angular/router/index.jade]: I've switched to sentence case.

File: src/angular/router/index.jade [hunk 14]
```diff
@@ -0,0 +1,312 @@
...
+//- TODO: mention `ActivatedRoute` and how to access it?
+
+:marked
+  ### Summary
+
+  The application has a configured router.
+  The shell component has a `RouterOutlet` where it can display views produced by the router.
+  It has `RouterLink`s that users can click to navigate via the router.
+
+  Here are the key `Router` terms and their meanings:
```
Comment (chalin) [inline src/angular/router/index.jade]: Changed to
> router

File: src/_data/side-nav.yml (hunk 5 above)
Comment (chalin) [inline src/_data/side-nav.yml]: Done

File: src/angular/router/index.jade (hunk 2 above)
Comment (chalin) [inline src/angular/router/index.jade]: Changed to "Routing".

File: src/angular/router/index.jade (hunk 3 above)
Comment (chalin) [inline src/angular/router/index.jade]: fixed

File: src/angular/router/index.jade [hunk 15]
```diff
@@ -0,0 +1,312 @@
...
+
+:marked
+  There are several flavors of `RouteDefinition`.
+  The most common, illustrated above, is the named **`Route`** which maps a URL path to a component.
+
+:marked
+  ### Router outlet
+
+  Given this configuration, when the browser URL for this application becomes `/heroes`,
+  the router matches that URL to the `RouteDefinition` named `Heroes` and displays the `HeroesComponent`
```
Comment (chalin) [inline src/angular/router/index.jade]: Opened https://github.com/dart-lang/site-webdev/issues/284

**Comment by chalin** [comment]
@matanlurey @kwalrath - this is ready for review. This first version of the page is mainly based on the TS router page _before_ they migrated to the new (which is current) router API. The material for Milestone 6 is still TBC, but most of the previous material is complete. If you are agreeable to the current version of the page, we can make iterative improvements (in particular for Milestone 6).

**Comment by kwalrath** [comment]
Taking off Matan, since we've requested reviews from @henrysenger and @potato620.

File: src/angular/guide/router/crisis-center-1.jade [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+    HTML 5 style navigation is the router default.
+    In the [Browser URL Styles](appendices#browser-url-styles) Appendix, 
+    learn why HTML 5 style is preferred, how to adjust its behavior, and how to switch to the
+    older hash (#) style, if necessary.
+
+:marked
+  You must add a [&lt;base href> element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base)
+  to the app's `index.html` for `pushState` routing to work.
+  The browser uses the base `href` value to prefix *relative* URLs when referencing
+  CSS files, scripts, and images.
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-1.jade]: Consider giving some examples here of how the router uses the base href. In the current documentation for the existing routers it isn't clear here what specifically changing the base href will do.

File: src/angular/guide/router/crisis-center-1.jade [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+:marked
+  ## @RouteConfig
+
+  A router must be configured with a list of route definitions.
+  A router also needs a *host component*, a point of origin for its navigations.
+
+  The `@RouteConfig` annotation combines the creation of a new router, its configuration, and
+  its assignment to a host component in a single step.
+
++makeExcerpt('app/app.component.1.ts', 'routes', '')
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-1.jade]: Probably a stupid question but why are we referencing Typescript files here? I thought this was the Dart documentation?

File: src/angular/guide/router/crisis-center-1.jade [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+  The router draws upon its registry of definitions when the browser URL changes
+  or when application code tells the router to navigate along a route path.
+
+  In simpler terms, you might say this of the first route:
+
+  - When the browser's location URL changes to match the path segment `/crisis-center`, then
+  the router activates an instance of the `CrisisCenterComponent` and displays its view.
+
+  - When the application requests navigation to the route named `CrisisCenter`, then the router
+  activates an instance of `CrisisCenterComponent`, displays its view, and updates the
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-1.jade]: What does "activates" mean exactly here? Will a new CrisisCenterComponent be instantiated every time I navigate to that URL? If I leave and come back, will the same component be reused?

File: src/angular/guide/router/crisis-center-1.jade [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+  A router must be configured with a list of route definitions.
+  A router also needs a *host component*, a point of origin for its navigations.
+
+  The `@RouteConfig` annotation combines the creation of a new router, its configuration, and
+  its assignment to a host component in a single step.
+
++makeExcerpt('app/app.component.1.ts', 'routes', '')
+
+:marked
+  The `@RouteConfig()` is applied to `AppComponent` which makes it the router's host component.
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-1.jade]: Consider stating that the host component must contain a router outlet.

File: src/angular/guide/router/crisis-center-1.jade [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+  * the name of the route that prescribes the destination component and a path for the URL
+  * optional route and query parameters that go into the route URL
+
+  The lists shown in the example above, each have a single parameter, a route name.
+  You'll see the use of route parameters later.
+
+//- TODO: mention routerLink: class.router-link-active here?
+
+:marked
+  ## *Declaring router providers and directives*
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-1.jade]: Consider moving this section up.

File: src/angular/guide/router/crisis-center-2.jade [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+
++makeExcerpt('lib/app_component_2.dart', 'Route.useAsDefault')
+
+:marked
+  **Refresh the browser** and try it. Notice that the heroes list is displayed when the application launches.
+
+:marked
+  ## Redirect route
+
+  As an alterivate solution, remove the `useAsDefault` argument that you just added,
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-2.jade]: s/alterivate/alternative/

File: src/angular/guide/router/crisis-center-2.jade [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+  and instead add a `Redirect` route that will translate an initial relative URL (`''`)
+  to the desired default path (`/heroes`).
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-2.jade]: Do redirects happen with a pushState or replaceState?

File: src/angular/guide/router/crisis-center-2.jade [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+  A _wildcard_ route is a route with a path starting with `/*`, followed by at least one character.
+  Such a route matches _every_ URL.
+
+.l-sub-section
+  :marked
+    The substring following the initial `/*` in a wildcard path defines the path segment name.
+    By convention, the path `/**` is used, resulting in a path segment name of just `*`.
+
+:marked
+  The router will select a wildcard route if it can't match a route earlier in the configuration.
```
Comment (henrysenger) [inline src/angular/guide/router/crisis-center-2.jade]: Maybe too in-depth for this tutorial, but consider explaining how the router must match the *entire* path (and not a prefix of it) in order for it to count as a match.

File: src/angular/guide/router/crisis-center-1.jade (hunk 17 above)
Comment (chalin) [inline src/angular/guide/router/crisis-center-1.jade]: Most doc pages use a form of "subclassing" to create the Dart docs. That is, the base class is the TS doc and we just write a minimal Dart doc subclass. Under this scheme, the Jade mixin (`+makeExcerpt`) handles the path translation from TS to Dart. This is part of what makes the subclassing scheme work. (The mixin does other work like deriving the title from the path and dropping file variant numeric suffixes in the process.)

Just like in programming, subclassing reduces the effort of creating the Dart-side docs. But since the split we've been breaking away from this scheme. So eventually, TS example paths and other such artifacts from times past, will be cleaned out/replaced with Dart-only prose elements.

**Comment by kwalrath** [comment]
Thanks for the review, @henrysenger. We expect @chalin to be able to work on this doc next week.

File: src/angular/guide/router/crisis-center-3.jade [hunk 24]
```diff
@@ -0,0 +1,368 @@
...
+:marked
+  You may sometimes wish to add *optional* information to a route request.
+  For example, when returning to the heroes list from the hero detail view, 
+  it might be nice if the viewed hero was preselected in the list.
+
+figure.image-display
+  img(src='/resources/images/devguide/router/selected-hero.png' alt="Selected hero")
+
+:marked
+  This becomes possible if you can include the viewed hero's `id` in the URL when
```
Comment (potato620) [inline src/angular/guide/router/crisis-center-3.jade]: mention somewhere that an optional parameter is a query param and link to an online resource

File: src/angular/guide/router/crisis-center-3.jade [hunk 25]
```diff
@@ -0,0 +1,368 @@
...
+    With this update to `goBack`, the `Location` service is no longer needed. Consider removing the
+    `_location` class field, corresponding constructor parameter and `Location` import.
+
+:marked
+  **Refresh the browser**, select a hero and then click the _Back_ button to return to the heroes list.
+  Notice that the URL now ends with a query parameter like this: `.../heroes?id=15`.
+
+.l-sub-section
+  :marked
+    The router can also encode parameters using the
```
Comment (potato620) [inline src/angular/guide/router/crisis-center-3.jade]: IMO seems unnecessary to mention this here

File: src/angular/guide/router/crisis-center-4.jade [hunk 26]
```diff
@@ -0,0 +1,449 @@
...
+  - *Root* of the crisis center area,
+  just as `AppComponent` is the root of the entire application
+  - *Shell* for the crisis management feature area,
+  just as the `AppComponent` is a shell to manage the high-level workflow
+
+  Like most shells, the `CrisisCenterComponent` class is very simple, simpler even than `AppComponent`:
+  it has no business logic, and it's template has no links, just a title and 
+  `<router-outlet>` for the crisis center child views.
+
+  Unlike `AppComponent`, and most other components, it _lacks a selector_.
```
Comment (potato620) [inline src/angular/guide/router/crisis-center-4.jade]: There is still a selector ATM

File: src/angular/guide/router/crisis-center-4.jade [hunk 27]
```diff
@@ -0,0 +1,449 @@
...
+  * There is no `crisis-center-home.component.ts`, and so
+  * no `<router-outlet>` in the crises component.
+  * Consequently, crisis details are rendered directly as a child view of `CrisisCenterComponent`.
+//- Note: we haven't yet made the crisis center the default.
+
+:marked
+  These two routes navigate to the crisis center child components,
+  `CrisisCenterHomeComponent` and `CrisisDetailComponent`, respectively.
+
+  Being _child routes_, there are some *important differences* in the way the router treats them:
```
Comment (potato620) [inline src/angular/guide/router/crisis-center-4.jade]: "important differences", but only one is listed. Maybe also mention that the paths are relative to the CrisisCenterComponent

ie: the /:id is actually /crisis-center/:id

**Comment by kwalrath** [comment]
Thanks for the review, @potato620. I'll leave this to @chalin to answer when he gets back.

File: src/angular/guide/router/index.jade [hunk 28]
```diff
@@ -0,0 +1,315 @@
...
++makeExcerpt('index.html', 'base-href')
+
+:marked
+  ### Router import
+
+  The Angular router is not part of the Angular core.
+  It is an optional service packaged within its own library, `angular2.router`.
+  Import it like any other package.
+
++makeExcerpt('lib/app_component_1.dart', 'import', '')
```
Comment (chalin) [inline src/angular/guide/router/index.jade]: By default `+makeExcerpt` creates a code excerpt with a title (which matches the first argument), a blank 3rd argument instructs the mixin to drop the title, and therefore only show the code.

File: src/angular/guide/router/crisis-center-2.jade (hunk 21 above)
Comment (chalin) [inline src/angular/guide/router/crisis-center-2.jade]: lol, wow, thanks for spotting that.
Fixed

File: src/_data/side-nav.yml [hunk 29]
```diff
@@ -67,8 +67,18 @@
...
+          children:
+            - title: "Routing: Overview"
+              permalink: /angular/guide/router
+              urlExactMatch: true
+            - title: "Routing 1: Basics"
+              permalink: /angular/guide/router/crisis-center-1
+            - title: "Routing 2: Hero Feature"
+              permalink: /angular/guide/router/crisis-center-2
+            - title: "Routing 3: Crisis Center Feature"
+              permalink: /angular/guide/router/crisis-center-3
```
Comment (chalin) [inline src/_data/side-nav.yml]: Actually, none of those sub-sub-menu items should be there because the side nav doesn't show entries to that depth.
Fixed

File: src/angular/guide/router/index.jade [hunk 30]
```diff
@@ -0,0 +1,315 @@
...
+  When the browser's URL changes, the router looks for a corresponding **`RouteDefinition`**
+  from which it can determine the component to display.
+
+  A router has no routes until you configure it.
+  The following example creates some route definitions. It illustrates
+  the preferred way of simultaneously creating a router and 
+  adding its routes using a **`@RouteConfig`**
+  applied to the router's host component:
+
++makeExcerpt('app/app.component.1.ts (routes)', 'app-component-routes')
```
Comment (chalin) [inline src/angular/guide/router/index.jade]: I'm not sure I understand this request. Are you asking for a hyperlink to the file? We don't actually publish such transitory files -- i.e., files with numbers in their names.  Such files _are_ accessible via the docs repo, but we don't link to them, instead we direct readers to the official samples collected under https://github.com/angular-examples.

File: src/angular/guide/router/crisis-center-3.jade [hunk 31]
```diff
@@ -0,0 +1,368 @@
...
+    - `hero_service.dart`
+    - `heroes_component.*`, the CSS, Dart and HTML files; <br>
+      **Note**: this will overwrite the local Dart file.
+    - `mock_heroes.dart`
+  - Use your IDE to refactor/move all of these newly copied files into the `lib/heroes` folder.
+  - In `heroes_component.html`, rename the `<ul>` element class to `items` (from `heroes`).
+  - Add the `HeroService` to the `providers` list of `AppComponent`
+    so that it is available everywhere in the app. This should be the result:
+
++makeExcerpt('lib/app_component_3.dart (v1)')
```
Comment (chalin) [inline src/angular/guide/router/crisis-center-3.jade]: The `v1` names the region being shown here. This could also have been written as:
`+makeExcerpt('lib/app_component_3.dart', 'v1')`.

File: src/angular/guide/router/index.jade (hunk 30 above)
Comment (kwalrath) [inline src/angular/guide/router/index.jade]: If you want to see the source code, the best way is probably to go to the rendered page and search for "(routes)".

@chalin, where is the rendered page?

File: src/angular/guide/router/crisis-center-3.jade [hunk 32]
```diff
@@ -0,0 +1,368 @@
...
+
+:marked
+  ## Route definition with a parameter
+
+  The route to `HeroDetailComponent` has a twist.
+  Notice the `:id` token at the end of the path;
+  that creates a slot in the path for a **Route Parameter**.
+  In this case, the router will insert the `id` of a hero into that slot.
+
+  If you tell the router to navigate to the detail component and display "Magneta",
```
Comment (chalin) [inline src/angular/guide/router/crisis-center-3.jade]: done

File: src/angular/guide/router/crisis-center-4.jade (hunk 26 above)
Comment (chalin) [inline src/angular/guide/router/crisis-center-4.jade]: Yes, there is still a selector, but selector's should not be mandatory, so I'd rather keep the prose as is for now and leave the warning in the code excerpt, directing the reader to  https://github.com/dart-lang/angular2/issues/60.

File: src/angular/guide/router/index.jade (hunk 30 above)
Comment (chalin) [inline src/angular/guide/router/index.jade]: Sorry, were you asking that a link be added in the prose, or that you simply wanted to see this transient file's content? If the latter, then see
- https://github.com/dart-lang/site-webdev/pull/283/files#diff-d8a501436c8079db9f220c459130a937 or
- https://github.com/chalin/site-webdev/blob/e1be6e9472bb346464e91b409fa10d69ba868594/public/docs/_examples/router/dart/lib/app_component_1.dart

File: src/angular/guide/router/crisis-center-4.jade (hunk 27 above)
Comment (chalin) [inline src/angular/guide/router/crisis-center-4.jade]: There are actually two bullets (though that is a bit harder to see from the .jade source), and the second bullet kind of makes the point you raised:
> screen shot 2017-02-07 at 2 29 43 pm

**Comment by kwalrath** [comment]
@chalin I've staged this locally and am looking at the sample, and I noticed a few issues:

1. The text and image show Heroes as being the first screen, but it displays Crisis Center.

image

2. As the screenshot shows, I see docregion comments in the UI (in both the crisis and hero screens).

3. The text says that selecting a hero takes you to an editing screen, but it doesn't.

4. There's no indication, from the buttons, of which screen you're looking at. The screenshot, on the other hand, implies that the current screen's button is highlighted.

5. The UI of the crisis editor has an extra button not shown in the screenshot. Also, the order of the buttons is different.

6. The text implies that you can see the crisis list at the same time as the crisis editor, but you can't.

7. http://localhost:4001/angular/guide/router/crisis-center-6 has **[TBC]** at the end. What should it really have?

I'll take a more careful look at the words tomorrow.

**Comment by chalin** [comment]
> 1. The text and image show Heroes as being the first screen, but it displays Crisis Center.

Fixed prose and code (that behavior was from the old version of the AngularDart router draft).

> 2. As the screenshot shows, I see docregion comments in the UI (in both the crisis and hero screens).

Those don't appear in the [published example](https://angular-examples.github.io/router), only when building and running from the "raw" sources contained under webdev. But when a reader is walking through the material he/she won't be typing docregion tags and so it won't appear for them either.
> 3. The text says that selecting a hero takes you to an editing screen, but it doesn't.

Fixed (this was due to test code I had added for the next parts).

> 4. There's no indication, from the buttons, of which screen you're looking at. The screenshot, on the other hand, implies that the current screen's button is highlighted.

Correct, that is not implemented yet. It is part of my to do list to add in the next round of changes.

Your other points (5, 6, & 7) are, as indicated TBC.  As I pointed out before, this is a _very_ long guide and I'd much prefer adopting an iterative approach to its publication. Part of the challenge is that features are handled differently in NgDart as compared to NgTS.

**Comment by kwalrath** [comment]
> Your other points (5, 6, & 7) are, as indicated TBC. As I pointed out before, this is a very long guide and I'd much prefer adopting an iterative approach to its publication. Part of the challenge is that features are handled differently in NgDart as compared to NgTS.

I'm fine with an iterative approach, but we need to be clear to readers about what they can expect. If they find errors and inconsistencies, they won't know what parts of the doc, if any, they can trust. The clearer we can be, the better.

**Comment by chalin** [comment]
For we can mark the entire router guide as having a "DRAFT" status. On my next pass, I can take note of which parts are complete and which parts have known issues (and then identifying the specific issues in the text via, say, callouts). WDYT?

**Comment by kwalrath** [comment]
After a few small changes, I'm fine with releasing this as a draft.

1. Add a prominent note to /angular/guide/router that this is a draft, perhaps soliciting feedback. Maybe, in a green(?) box:
   
   **This is a draft.** The router pages are still being updated. We've published them so you can get the information you need and [give us feedback](https://github.com/dart-lang/site-webdev/issues/new).
 
1. /angular/guide/router should point to /angular/guide/pipes as the previous page. (Pipes correctly has router as the next page.)

1. Change the "[TBC]" in crisis-center-6 to something a little clearer. Perhaps just **Under construction...**

**Comment by chalin** [comment]
@kwalrath - I've addressed the items you mentioned in the previous comment.

**Comment by kwalrath** [comment]
LGTM. We'll iterate!

**Comment by chalin** [comment]
@henrysenger @potato620 - just to let you know that I addressed your most pressing concerns in this PR. The other points that you raised will be addressed in followup PR.

**Comment by potato620** [comment]
sgtm 👍🏻
